astropy/table/meta.py

```diff
@@ -236,59 +127,12 @@ def get_yaml_from_header(header):
 
     from astropy.io.misc.yaml import AstropyDumper
 
-    class TableDumper(AstropyDumper):
-        """
-        Custom Dumper that represents OrderedDict as an !!omap object.
-        """
-
-        def represent_mapping(self, tag, mapping, flow_style=None):
-            """
-            This is a combination of the Python 2 and 3 versions of this method
-            in the PyYAML library to allow the required key ordering via the
-            ColumnOrderList object.  The Python 3 version insists on turning the
-            items() mapping into a list object and sorting, which results in
-            alphabetical order for the column keys.
-            """
-            value = []
-            node = yaml.MappingNode(tag, value, flow_style=flow_style)
-            if self.alias_key is not None:
-                self.represented_objects[self.alias_key] = node
-            best_style = True
-            if hasattr(mapping, 'items'):
-                mapping = mapping.items()
-                if hasattr(mapping, 'sort'):
-                    mapping.sort()
-                else:
-                    mapping = list(mapping)
-                    try:
-                        mapping = sorted(mapping)
-                    except TypeError:
-                        pass
-
-            for item_key, item_value in mapping:
-                node_key = self.represent_data(item_key)
-                node_value = self.represent_data(item_value)
-                if not (isinstance(node_key, yaml.ScalarNode) and not node_key.style):
-                    best_style = False
-                if not (isinstance(node_value, yaml.ScalarNode) and not node_value.style):
-                    best_style = False
-                value.append((node_key, node_value))
-            if flow_style is None:
-                if self.default_flow_style is not None:
-                    node.flow_style = self.default_flow_style
-                else:
-                    node.flow_style = best_style
-            return node
-
-    TableDumper.add_representer(OrderedDict, _repr_odict)
-    TableDumper.add_representer(ColumnDict, _repr_column_dict)
-
-    header = copy.copy(header)  # Don't overwrite original
     header['datatype'] = [_get_col_attributes(col) for col in header['cols']]
+    header = {key: header[key] for key in sorted(header)}
```

Comment:
It doesn't *need* to be sorted, but it does keep the ECSV header consistent with past behavior and also tidier going forward. I.e. if we change the code so that keys get added in a different order then the output will be the same. I noticed this of course because of a regression fail without the `sorted`.

Reply:
OK, I mostly wondered if it was already done automatically, but you had already checked. Agree there is no need to change the behaviour.